applicationconfigurations/workflow.md

```diff
@@ -0,0 +1,219 @@
+# An Example Workflow
+
+This section is non-normative. OAM compliant tooling need not implement this section.
+
+This section uses a fictional tool called `oamctl` to illustrate a workflow pattern for OAM app operations.
+
+### Deploying two components with parameter overrides
+The following example shows two separate components:
+ - a front-end web app in a container, run as a core container workload.
+ - a back-end Cassandra database, represented by an extended workload.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: frontend
+  annotations:
+    version: v1.0.0
+    description: "A simple webserver"
+spec:
+  workload:
+    apiVersion: core.oam.dev/v1alpha2
+    kind: containerWorkload
+    metadata:
+      name: sample-workload
+    spec:
+      osType: linux
+      containers:
+      - name: web
+        image:
+          name: example/charybdis-single:latest
+        resources:
+          cpu:
+            required: 1.0
+          memory:
+            required: 100MB
+        env:
+        - name: MESSAGE
+          value: default
+  parameters:
+  - name: message
+    description: The message to display in the web app.  
+    required: true
+    type: string
+    value: "Hello from my app, too"
+    fieldPaths:
+    - "spec.containers[0].env[0].value"
+```
+
+```yaml
+apiVersion: core.oam.dev/v1alpha1
+kind: Component
+metadata:
+  name: backend
+  annotations:
+    version: v1.0.0
+    description: "Cassandra database"
+spec:
+  workload:
+    apiVersion: data.oam.dev/v1
+    kind: Cassandra
+    spec:
+      maxStalenessPrefix: 100000
+      defaultConsistencyLevel: Eventual
+  parameters:
+    - name: maxStalenessPrefix
+      description: Max stale requests.
```

Comment:
```suggestion
```

Reply:
Why do we need to delete description?

Reply:
This is a result of its removal in #316 https://github.com/oam-dev/spec/pull/316/files#diff-69868a64b8b4c463fcecee89ee0da1c7R44

It sounds like you are planning on waiting until that it merged anyway so it may make sense to just wait until changes there are stable :)

Reply:
It looks like we are planning to keep `description` now: https://github.com/crossplane/crossplane/pull/1276#discussion_r384886178